lib/api_base_rest/cache/cache.dart

```diff
@@ -0,0 +1,35 @@
+import "dart:convert";
+import "dart:typed_data";
+
+import "package:flutter_cache_manager/flutter_cache_manager.dart";
+import "package:flutter_riverpod/flutter_riverpod.dart";
+import "../../shared_api_clients/sks_api_client.dart";
+import "cache_manager.dart";
+
+Future<T> getAndCacheData<T>(
+  String url,
+  Ref ref,
```

Comment:
did you consider making it an extension on `Ref` instead of a normal function?

Reply:
Yes that's good Idea I think